File: frontend/gestao/src/entities/historico-login/model/historico-login.dto.ts
import {z} from 'zod';

export const historicoLoginDtoSchema = z.object({
	id: z.string().uuid(),
	identityId: z.string(),
	usuarioId: z.string().uuid().nullable().optional(),
	tipo: z.string(),
	relacionadoId: z.string().nullable().optional(),
	email: z.string().nullable().optional(),
	nome: z.string().nullable().optional(),
	ip: z.string().nullable().optional(),
	userAgent: z.string().nullable().optional(),
	sucesso: z.boolean(),
	dataEvento: z.string(),
	sessaoBffId: z.string().nullable().optional(),
});

export type HistoricoLoginDto = z.infer<typeof historicoLoginDtoSchema>;

File: spring-boot/backend/domain-security/src/main/java/com/lodh8/starter/security/historico/api/dto/HistoricoLoginDTO.java
package com.lodh8.starter.security.historico.api.dto;

import com.lodh8.starter.shared.models.enums.TipoAcesso;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HistoricoLoginDTO {

    private UUID id;

    private String identityId;

    private UUID usuarioId;

    private TipoAcesso tipo;

    private String relacionadoId;

    private String email;

    private String nome;

    private String ip;

    private String userAgent;

    private boolean sucesso;

    private OffsetDateTime dataEvento;

    private String sessaoBffId;

}
